src/kerberos_sspi.c

```diff
@@ -358,6 +437,119 @@ auth_sspi_client_step(sspi_client_state* state, SEC_CHAR* challenge, SecPkgConte
     return status;
 }
 
+INT
+auth_sspi_server_step(sspi_server_state* state, SEC_CHAR* challenge) {
+	SecBufferDesc inbuf;
+	SecBuffer inBufs[2];
+	SecBufferDesc outbuf;
+	SecBuffer outBufs[1];
+	ULONG ignored;
+	SECURITY_STATUS status = AUTH_GSS_CONTINUE;
+	DWORD len;
+	BOOL haveToken = FALSE;
+	INT tokenBufferIndex = 0;
+
+	if (state->response != NULL) {
+		free(state->response);
+		state->response = NULL;
+	}
+
+	inbuf.ulVersion = SECBUFFER_VERSION;
+	inbuf.pBuffers = inBufs;
+	inbuf.cBuffers = 0;
+
+	tokenBufferIndex = inbuf.cBuffers;
+	haveToken = TRUE;
+	inBufs[tokenBufferIndex].BufferType = SECBUFFER_TOKEN;
+	inBufs[tokenBufferIndex].pvBuffer = base64_decode(challenge, &len);
+	if (!inBufs[tokenBufferIndex].pvBuffer) {
+		return AUTH_GSS_ERROR;
+	}
+	inBufs[tokenBufferIndex].cbBuffer = len;
+	inbuf.cBuffers++;
+
+	/* Get cbMaxToken size for the output buffer */
+	PSecPkgInfo pkgInfo;
```

Comment:
After what seems like forever I'm finally back working on this. I think the reason you're getting a memory error might be related to this chunk of code. If you don't tell AcceptSecurityContext to allocate buffers for you then you have to allocate the out buffer yourself. The flag to use is ASC_REQ_ALLOCATE_MEMORY (rather than ISC_REQ_ALLOCATE_MEMORY in InitializeSecurityContext). Microsoft doesn't make this super easy to find. I'm working on fixing this up myself.

https://docs.microsoft.com/en-us/windows/desktop/api/sspi/nf-sspi-acceptsecuritycontext